### PR TITLE
doc/rados: edit troubleshooting-osd (1 of x)

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/rados/troubleshooting/troubleshooting-osd.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 1]
```diff
@@ -2,202 +2,260 @@
...
+- **Check ``nf_conntrack``:** This connection-tracking and connection-limiting
+  system causes problems for many production Ceph clusters. The problems often
+  emerge slowly and subtly. As cluster topology and client workload grow,
+  mysterious and intermittent connection failures and performance glitches
+  occur more and more, especially at certain times of the day. To begin taking
+  the measure of your problem, check the ``syslog`` history for "table full"
+  events. One way to address this kind of problem is as follows: First, use the
+  ``sysctl`` utility to assign ``nf_conntrack_max`` a much higher value. Next,
+  raise the value of ``nf_conntrack_buckets`` so that ``nf_conntrack_buckets``
+  × 4 = ``nf_conntrack_max``; this action might require running commands
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:230]: I goofed here, this should be x8 not x4.

**Review by anthonyeleven — APPROVED** [review]

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 2]
```diff
@@ -2,202 +2,260 @@
...
+dropped packets on the host side and CRC errors on the switch side.
+
 
 Obtaining Data About OSDs
 =========================
 
-A good first step in troubleshooting your OSDs is to obtain topology information in
-addition to the information you collected while `monitoring your OSDs`_
-(e.g., ``ceph osd tree``).
+To begin troubleshooting OSDs, it is useful to collect different kinds of
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:24]: Yeah, that is better.

Accepted.

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 3]
```diff
@@ -2,202 +2,260 @@
...
+Additional information concerns the topology of your cluster, and is discussed
+in the following sections.
 
 
 Ceph Logs
 ---------
 
-If you haven't changed the default path, you can find Ceph log files at
-``/var/log/ceph``::
+Ceph log files are stored at a default location of ``/var/log/ceph``. Unless
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:34]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 4]
```diff
@@ -2,202 +2,260 @@
...
+- Listing Ceph configuration at runtime
+- Dumping historic operations
+- Dumping the operation priority queue state
+- Dumping operations in flight
+- Dumping perfcounters
+
+Display Free Space
+------------------
+
+Filesystem issues may arise. To display your filesystem's free space, run the
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:83]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 5]
```diff
@@ -2,202 +2,260 @@
...
+in the following sections.
 
 
 Ceph Logs
 ---------
 
-If you haven't changed the default path, you can find Ceph log files at
-``/var/log/ceph``::
+Ceph log files are stored at a default location of ``/var/log/ceph``. Unless
+the path has been changed, the log files can be listed by running the following
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:35]: Accepted with a slight modification.

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 6]
```diff
@@ -2,202 +2,260 @@
...
+    dmesg | grep scsi
+
+Stopping without Rebalancing
+============================
+
+It might be occasionally necessary to perform maintenance on a subset of your
+cluster or to resolve a problem that affects a failure domain (for example, a
+rack).  However, when you stop OSDs for maintenance, you might want to prevent
+CRUSH from automatically rebalancing the cluster. To avert this rebalancing
+behavior, set the cluster to ``noout`` by running the following command:
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:121]: I've added a fairly beefy note to that effect.

Accepted.

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 7]
```diff
@@ -2,202 +2,260 @@
...
+   systemctl stop ceph\*.service ceph\*.target
 
-Once you have completed your maintenance, restart the OSDs and any other
-daemons.  If you rebooted the host as part of the maintenance, these should
-come back on their own without intervention. ::
+.. note:: When an OSD is stopped, any placement groups within the OSD are
+   marked as ``degraded``.
 
-	sudo systemctl start ceph.target
+After the maintenance is complete, it might be necessary to restart the OSDs
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:152]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 8]
```diff
@@ -2,202 +2,260 @@
...
+  (usually 32,000).  This is especially likely to happen during recovery.
+  Increasing the maximum number of threads to the maximum possible number of
+  threads allowed (4194303) might help with the problem. To increase the number
+  of threads to the maximum, run the following command:
+
+  .. prompt:: bash
+
+     sysctl -w kernel.pid_max=4194303
+
+  If this increase resolves the issue, you might want to make the increase
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:214]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-osd.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:230]: I've changed "4" to "8" here.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53982 - Reef backport
https://github.com/ceph/ceph/pull/53983 - Quincy backport

**Pull request merged**